### PR TITLE
[MM-34491] Don't overwrite root command label

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Respect slash command root label instead of using the app ID.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34491

### CODE REVIEW

**Review by larkox — APPROVED** [review]

**Comment by DHaussermann** [comment]
@hanzei this seems to be working on mobile though the first deploy needs a debug-clean. 

The slash command binding is now working with the `display_name` and the App ID is no longer appearing. 
It may make sense to live this PR open until we see this behavior on webapp as well.  

Side note: Huge thanks for the addition of the auto-complete for using the debug commands. This is very helpful.

File: docs/01-anatomy-hello.md [hunk 1]
```diff
@@ -81,6 +81,7 @@ The Hello App creates a Channel Header button, and adds a `/helloworld send` com
 			"bindings": [
 				{
 					"icon": "http://localhost:8080/static/icon.png",
+                    "label": "helloworld",
```
Comment (mickmister) [inline docs/01-anatomy-hello.md:84]: Nit: indentation. Same with other json changes

**Review by mickmister — APPROVED** [review]

**Review by levb — APPROVED** [review]

File: server/proxy/bindings.go [hunk 2]
```diff
@@ -136,11 +136,6 @@ func (p *Proxy) scanAppBindings(app *apps.App, bindings []*apps.Binding, locPref
 			continue
 		}
 
-		if locPrefix == apps.LocationCommand {
```
Comment (levb) [inline server/proxy/bindings.go:139]: @hanzei Can we still default to AppID if the location/label are missing, or are you >0/5 on encouraging hierarchical AppIDs?

File: docs/01-anatomy-hello.md (hunk 1 above)
Comment (hanzei) [reply docs/01-anatomy-hello.md]: Good catch :+1:

File: server/proxy/bindings.go (hunk 2 above)
Comment (hanzei) [reply server/proxy/bindings.go:139]: I'm opposed to using the AppID by default for two reasons:
1. 3/5 I would like to think about the AppID as "a technical detail". It's a technical necessity and should not be user facing.
2. 1/5 Default values like that are not obvious. I would like to rather error with a meaningful message to developer then doing "magic".

**Comment by hanzei** [comment]
@levb I'm happy to continue the discussion about what default values we should set, but I'm going to merge this PR to unblock the other ones.

File: server/proxy/bindings.go (hunk 2 above)
Comment (levb) [reply server/proxy/bindings.go:139]: SG

**Pull request merged**